File: ibkr_show_backend/app/schemas/admin_email.py
from pydantic import BaseModel


class EmailSettingsResponse(BaseModel):
    enabled: bool
    smtp_host: str
    smtp_port: int
    smtp_username: str
    smtp_password_masked: str
    has_smtp_password: bool
    smtp_use_ssl: bool
    smtp_use_starttls: bool
    email_from: str
    email_to: str
    subject_prefix: str
    site_base_url: str
    config_file: str
    daily_review_email_enabled: bool
    daily_review_email_to: str
    daily_review_subject_prefix: str
    daily_snapshot_email_enabled: bool
    daily_snapshot_email_to: str
    daily_snapshot_subject_prefix: str


class EmailSettingsUpdateRequest(BaseModel):
    smtp_host: str
    smtp_port: int
    smtp_username: str
    smtp_password: str | None = None
    smtp_use_ssl: bool
    smtp_use_starttls: bool
    email_from: str
    daily_review_email_enabled: bool
    daily_review_email_to: str
    daily_review_subject_prefix: str | None = None
    site_base_url: str | None = None
    daily_snapshot_email_enabled: bool
    daily_snapshot_email_to: str
    daily_snapshot_subject_prefix: str | None = None


class EmailSettingsMutationResponse(BaseModel):
    settings: EmailSettingsResponse
    message: str


class EmailTestRequest(BaseModel):
    subject: str | None = None
    message: str | None = None


class EmailTestResponse(BaseModel):
    success: bool
    message: str
    sent_to: list[str]
    sent_at: str


class EmailSendLatestResponse(BaseModel):
    success: bool
    sent: bool
    report_date: str | None
    message: str
    task_id: str | None = None
    status: str | None = None


class EmailSendLatestDailyReviewRequest(BaseModel):
    force_refresh: bool = False
    regenerate_if_legacy: bool = True

File: ibkr_show_frontend/src/types/adminEmail.ts
export type EmailSettings = {
  enabled: boolean
  smtp_host: string
  smtp_port: number
  smtp_username: string
  smtp_password_masked: string
  has_smtp_password: boolean
  smtp_use_ssl: boolean
  smtp_use_starttls: boolean
  email_from: string
  email_to: string
  subject_prefix: string
  site_base_url: string
  config_file: string
  daily_review_email_enabled: boolean
  daily_review_email_to: string
  daily_review_subject_prefix: string
  daily_snapshot_email_enabled: boolean
  daily_snapshot_email_to: string
  daily_snapshot_subject_prefix: string
}

export type EmailSettingsPayload = {
  smtp_host: string
  smtp_port: number
  smtp_username: string
  smtp_password?: string
  smtp_use_ssl: boolean
  smtp_use_starttls: boolean
  email_from: string
  daily_review_email_enabled: boolean
  daily_review_email_to: string
  daily_review_subject_prefix?: string
  site_base_url?: string
  daily_snapshot_email_enabled: boolean
  daily_snapshot_email_to: string
  daily_snapshot_subject_prefix?: string
}

export type EmailSettingsMutationResponse = {
  settings: EmailSettings
  message: string
}

export type EmailTestPayload = {
  subject?: string
  message?: string
}

export type EmailTestResponse = {
  success: boolean
  message: string
  sent_to: string[]
  sent_at: string
}

export type EmailSendLatestResponse = {
  success: boolean
  sent: boolean
  report_date: string | null
  message: string
  task_id?: string | null
  status?: string | null
}
